@here/harp-materials/lib/CirclePointsMaterial.ts

```diff
@@ -43,46 +45,47 @@ export interface CirclePointsMaterialParameters extends THREE.ShaderMaterialPara
      * Point size.
      */
     size?: number;
-}
 
-const DEFAULT_CIRCLE_SIZE = 1;
+    /**
+     * Point color.
+     */
+    color?: THREE.Color;
+}
 
 /**
  * Material designed to render circle points. Note that it is always transparent since the circle
  * shape is created with an alpha channel to benefit an antialising that a mere `discard` could
  * not bring.
  */
 export class CirclePointsMaterial extends THREE.ShaderMaterial {
-    isCirclePointsMaterial: true;
-    uniforms: { [uniform: string]: THREE.IUniform };
-    vertexShader: string;
-    fragmentShader: string;
+    static readonly DEFAULT_CIRCLE_SIZE = 1;
 
-    private m_size: number;
     private m_color: THREE.Color;
+    private m_opacity: number;
 
     /**
      * Constructs a new `CirclePointsMaterial`.
      *
      * @param parameters The constructor's parameters.
      */
     constructor(parameters: CirclePointsMaterialParameters = {}) {
-        parameters.depthTest = false;
+        const { size, color, opacity, ...shaderParams } = parameters;
+        shaderParams.name = "CirclePointsMaterial";
+        shaderParams.vertexShader = vertexShader;
+        shaderParams.fragmentShader = fragmentShader;
+        shaderParams.depthTest = false;
 
-        super(parameters);
+        super(shaderParams);
         enforceBlending(this);
 
-        this.isCirclePointsMaterial = true;
         this.type = "CirclePointsMaterial";
-        this.vertexShader = vertexShader;
-        this.fragmentShader = fragmentShader;
-
-        this.m_size = parameters.size || DEFAULT_CIRCLE_SIZE;
-        this.m_color = new THREE.Color();
+        this.m_color = getOptionValue(color, new THREE.Color());
+        this.m_opacity = getOptionValue(opacity, 1.0);
```

Comment:
Just two side notes - make sure if it supports RGBA in technique (color assigned from technique should be already in RGB format) and opacity applied.... Maybe extend the test?
This should be OK, as generic algorithm is applied but could be good to double-check.

Reply:
The rgba support is done by `DecodedTileHelper`. The Alpha component is removed from the color there and passed as opacity. So there is no need to handle it in the materials.